File: CS/NetCore/net/08_SideButton/SideButton.cs
namespace InputManWin13_Demo._08_SideButton;

public partial class SideButton : InputManWin13_Demo.DemoBase_settings
{
    public SideButton()
    {
        InitializeComponent();
        InitializeForm();
    }

    private void InitializeForm()
    {
        // ***** 初期設定

        // CheckOnClickプロパティ
        checkBox1.CheckedChanged += (s, e) => SetCheckOnClick();

        // テキストコントロールのサイドボタン
        sideButton1.Click += (s, e) => ShowTextSideButtonPushMessage();

        // マスクコントロールのサイドボタン
        sideButton2.Click += (s, e) => ShowMaskSideButtonPushMessage();

        // 日付時刻コントロールのサイドボタン
        sideButton3.Click += (s, e) => SetNowToDateTimeValue();

        // 日付コントロールのサイドボタン
        sideButton4.Click += (s, e) => SetNowToDateValue();

        // タイムスパンコントロールのサイドボタン
        sideButton5.Click += (s, e) => SetStartOrStop();

        // 数値コントロールのサイドボタン（下）
        sideButton6.Click += (s, e) => SetNumberDown();

        // 数値コントロールのサイドボタン（上）
        sideButton7.Click += (s, e) => SetNumberUp();

        // コンボコントロールのサイドボタン
        sideButton8.Click += (s, e) => ClearComboBox();

        // マスクコンボコントロールのサイドボタン
        sideButton9.Click += (s, e) => ClearMask();
        timer1.Tick += (s, e) => SetTickTimer();

        // ***** 初期値
        checkBox1.Checked = ((GrapeCity.Win.Editors.SideButton)gcTextBox1.SideButtons[0]).CheckOnClick;

        timer1.Interval = 1000;

        // xmlファイルからデータ取得
        this.dataSet1.ReadXml(Path.Combine(Application.StartupPath, @"Resources\\prefecture.xml"));
        gcComboBox1.DataSource = dataSet1.Tables[0];
        gcComboBox1.SelectedIndex = -1;

        for (int i = 1; i < gcComboBox1.ListColumns.Count; i++)
        {
            gcComboBox1.ListColumns[i].AutoWidth = true;
        }

        // xmlファイルからデータ取得（gcMaskedComboBox）
        this.dataSet2.ReadXml(Path.Combine(Application.StartupPath, @"Resources\\pref.xml"));
        gcMaskedComboBox1.DataSource = dataSet2.Tables[0];
        gcMaskedComboBox1.SelectedIndex = -1;

        for (int i = 1; i < gcMaskedComboBox1.ListColumns.Count; i++)
        {
            gcMaskedComboBox1.ListColumns[i].AutoWidth = true;
        }
    }

    private void SetCheckOnClick()
    {
        ((GrapeCity.Win.Editors.SideButton)gcTextBox1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcMask1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcDateTime1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcDate1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcTimeSpan1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcNumber1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcNumber1.SideButtons[1]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcComboBox1.SideButtons[0]).CheckOnClick = checkBox1.Checked;
        ((GrapeCity.Win.Editors.SideButton)gcMaskedComboBox1.SideButtons[0]).CheckOnClick = checkBox1.Checked;

        if (!checkBox1.Checked)
        {
            ((GrapeCity.Win.Editors.SideButton)gcTextBox1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcMask1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcDateTime1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcDate1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcTimeSpan1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcNumber1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcNumber1.SideButtons[1]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcComboBox1.SideButtons[0]).Checked = false;
            ((GrapeCity.Win.Editors.SideButton)gcMaskedComboBox1.SideButtons[0]).Checked = false;
        }
    }

    private static void ShowTextSideButtonPushMessage() => MessageBox.Show("正解はTrueです！", "メッセージ", MessageBoxButtons.OK, MessageBoxIcon.Information);

    private static void ShowMaskSideButtonPushMessage() => MessageBox.Show("ここに住所検索処理を入れることができます");

    private void SetNowToDateTimeValue()
    {
        gcDateTime1.Value = DateTime.Now;
    }

    private void SetNowToDateValue()
    {
        gcDate1.Value = DateTime.Now;
    }

    private void SetStartOrStop()
    {
        if (timer1.Enabled)
        {
            timer1.Enabled = false;
            sideButton5.Text = "Start";
        }
        else
        {
            timer1.Enabled = true;
            sideButton5.Text = "Stop";
            if (gcTimeSpan1.Value == null)
            {
                gcTimeSpan1.Value = new TimeSpan(0);
            }
            gcTimeSpan1.Value = gcTimeSpan1.Value + new TimeSpan(0, 0, 0, 1);
        }
    }

    private void SetNumberDown()
    {
        if (gcNumber1.Value > 0)
        {
            gcNumber1.Value -= 1;
        }
    }

    private void SetNumberUp()
    {
        if (gcNumber1.Value < 999)
        {
            gcNumber1.Value += 1;
        }
    }

    private void ClearComboBox()
    {
        gcComboBox1.SelectedIndex = -1;
        gcComboBox1.Clear();
    }

    private void ClearMask()
    {
        gcMaskedComboBox1.SelectedIndex = -1;
        gcMaskedComboBox1.Clear();
    }
    private void SetTickTimer()
    {
        gcTimeSpan1.Value = gcTimeSpan1.Value + new TimeSpan(0, 0, 0, 1);
    }
}

File: VB/NetCore/net/08_SideButton/SideButton.vb
Imports System.Windows.Forms

Namespace _08_SideButton
	Public Partial Class SideButton
        Inherits InputManWin13_Demo.DemoBase_settings
        Public Sub New()
			InitializeComponent()
			InitializeForm()
		End Sub

		Private Sub InitializeForm()
			' ***** 初期値
			checkBox1.Checked = DirectCast(gcTextBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick

			timer1.Interval = 1000

			' xmlファイルからデータ取得
			Dim filePath As System.String = Application.StartupPath
			Me.dataSet1.ReadXml(filePath & "\Resources\prefecture.xml")
			gcComboBox1.DataSource = dataSet1.Tables(0)
			gcComboBox1.SelectedIndex = -1

			For i As Integer = 1 To gcComboBox1.ListColumns.Count - 1
				gcComboBox1.ListColumns(i).AutoWidth = True
            Next

            ' xmlファイルからデータ取得(GcMaskedComboBox)
            Me.DataSet2.ReadXml(filePath & "\Resources\pref.xml")
            GcMaskedComboBox1.DataSource = DataSet2.Tables(0)
            GcMaskedComboBox1.SelectedIndex = -1

            For i As Integer = 1 To GcMaskedComboBox1.ListColumns.Count - 1
                GcMaskedComboBox1.ListColumns(i).AutoWidth = True
            Next

		End Sub

        Private Sub checkBox1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBox1.CheckedChanged
            ' CheckOnClickプロパティ
            DirectCast(gcTextBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcMask1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcDateTime1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcDate1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcTimeSpan1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcNumber1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcNumber1.SideButtons(1), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(gcComboBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked
            DirectCast(GcMaskedComboBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).CheckOnClick = checkBox1.Checked

            If Not checkBox1.Checked Then
                DirectCast(gcTextBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcMask1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcDateTime1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcDate1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcTimeSpan1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcNumber1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcNumber1.SideButtons(1), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(gcComboBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
                DirectCast(GcMaskedComboBox1.SideButtons(0), GrapeCity.Win.Editors.SideButton).Checked = False
            End If
        End Sub

        Private Sub sideButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton1.Click
            ' テキストコントロールのサイドボタン
            MessageBox.Show("正解はTrueです！", "メッセージ", MessageBoxButtons.OK, MessageBoxIcon.Information)
        End Sub

        Private Sub sideButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton2.Click
            ' マスクコントロールのサイドボタン
            MessageBox.Show("ここに住所検索処理を入れることができます")
        End Sub

        Private Sub sideButton3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton3.Click
            ' 日付時刻コントロールのサイドボタン
            gcDateTime1.Value = DateTime.Now
        End Sub

        Private Sub sideButton4_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton4.Click
            ' 日付コントロールのサイドボタン
            gcDate1.Value = DateTime.Now
        End Sub

        Private Sub sideButton5_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton5.Click
            ' タイムスパンコントロールのサイドボタン
            If timer1.Enabled Then
                timer1.Enabled = False
                sideButton5.Text = "Start"
            Else
                timer1.Enabled = True
                sideButton5.Text = "Stop"
                If gcTimeSpan1.Value Is Nothing Then
                    gcTimeSpan1.Value = New TimeSpan(0)
                End If
                gcTimeSpan1.Value = gcTimeSpan1.Value + New TimeSpan(0, 0, 0, 1)
            End If
        End Sub

        Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs) Handles timer1.Tick
            gcTimeSpan1.Value = gcTimeSpan1.Value + New TimeSpan(0, 0, 0, 1)
        End Sub

        Private Sub sideButton6_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton6.Click
            ' 数値コントロールのサイドボタン（下）
            If gcNumber1.Value > 0 Then
                gcNumber1.Value -= 1
            End If
        End Sub

        Private Sub sideButton7_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton7.Click
            ' 数値コントロールのサイドボタン（上）
            If gcNumber1.Value < 999 Then
                gcNumber1.Value += 1
            End If
        End Sub

        Private Sub sideButton8_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton8.Click
            ' コンボコントロールのサイドボタン
            gcComboBox1.SelectedIndex = -1
            gcComboBox1.Clear()
        End Sub

        Private Sub SideButton9_Click(sender As Object, e As EventArgs) Handles SideButton9.Click
            ' マスクコンボコントロールのサイドボタン
            GcMaskedComboBox1.SelectedIndex = -1
            GcMaskedComboBox1.Clear()
        End Sub
    End Class
End Namespace
